portality/events/consumer_utils.py

```diff
@@ -0,0 +1,23 @@
+from portality import models
+from portality.bll import exceptions
+from portality.core import app
+
+
+def parse_application(application_raw: dict) -> models.Application:
+    try:
+        return models.Application(**application_raw)
+    except Exception as e:
+        raise exceptions.NoSuchObjectException(
+            "Unable to construct Application from supplied source - data structure validation error, {x}".format(x=e)
+        )
```

Comment:
This should be in the `portality.bll.application.ApplicationService` object called something like `construct_application_from_raw` or similar

Reply:
Not recommend to make this method to public and general.
each components might have different ways to handle exception.

convert to `NoSuchObjectException` with specific message is fine for consumer module if they want to handle all exception with that message. But this convert will loss information of lower level exception and loss flexibility to handle other exceptions in different ways.

This method will stay in `consumer_utils` for now, since no other module would convert exception in this way.

Reply:
Ok, I'm happy to leave it for now, but I feel that probably there's a better reorganisation of code that we haven't yet thought of.

For example, the `NoSuchObjectException` is probably wrong in this case, anyway, as that's a BLL level exception meaning that the object isn't in the database.  This is a pre-existing oddity, not added by this PR.  But perhaps a better consumer-level exception would be suitable here instead, OR we add a more general application-level exception which says that an object could not be parsed, and which is more general than the SeamlessException that this wraps.

In any case, a discussion for another time!